File: Creational/Builder/GoLang/meal_director.go
package main

type MealDirector struct {
	builder MealBuilder
}

func NewMealDirector(builder MealBuilder) *MealDirector {
	return &MealDirector{builder: builder}
}

func (d *MealDirector) ConstructMeal() *Meal {
	d.builder.BuildMainItem()
	d.builder.BuildSideItem()
	d.builder.BuildDrink()
	d.builder.BuildToy()
	return d.builder.GetMeal()
}

File: Creational/Builder/JavaScript/js/MealDirector.js
// js/MealDirector.js
export class MealDirector {
    constructor(builder) {
        this.builder = builder;
    }

    constructMeal() {
        this.builder.buildMainItem();
        this.builder.buildSideItem();
        this.builder.buildDrink();
        this.builder.buildToy();
        return this.builder.getMeal();
    }
}
